x-pack/plugins/security_solution/server/lib/detection_engine/signals/search_after_bulk_create.ts

```diff
@@ -118,7 +107,6 @@ export const searchAfterAndBulkCreate = async ({
     interval,
     buildRuleMessage,
   });
-  const useSortIds = totalToFromTuples.length <= 1;
```

Comment:
This line was the culprit of the bug that caused this revert https://github.com/elastic/kibana/pull/71956 . 

`useSortIds` was only > 1 when there was a gap, which would mean boolean `useSortIds` was false. This bug showed up because I modified the count we used to determine when we hit max signals from the number of search results to the number of bulk create results (signals indexed) with this pr https://github.com/elastic/kibana/pull/71768. When I did that, one of the circuit breakers for jumping out of the loop relied on checking the next sort id would fail because useSortIds was false. I’ve removed the variable and simplified the logic for breaking out of the while loop. That variable was leftover from a different implementation I attempted while doing the gap detection fixes and I missed deleting it and updating the logic.

Again this bug stayed hidden for so long after all the gap detection code was added because I never changed the count used to determine when we hit max signals until this commit went in https://github.com/elastic/kibana/commit/56de45d156be23069815fec17440cf978710451f right before FF.